fast/events/ios/pdf-modifer-key-down-crash.html leaves an unfinished UIScript running

smfr opened the pull request:
#### c1d267625fe474c6a0f475cfe5ffa34f837f9b9a
<pre>
fast/events/ios/pdf-modifer-key-down-crash.html leaves an unfinished UIScript running
<a href="https://bugs.webkit.org/show_bug.cgi?id=243332">https://bugs.webkit.org/show_bug.cgi?id=243332</a>
&lt;rdar://98164628&gt;

Reviewed by Ryosuke Niwa.

Use async/await in the queued script so that it waits for the runUIScript() to complete.

Also unskip the test.

* LayoutTests/fast/events/ios/pdf-modifer-key-down-crash.html:

Canonical link: <a href="https://commits.webkit.org/253253@main">https://commits.webkit.org/253253@main</a>
</pre>

rniwa commented on LayoutTests/fast/events/ios/pdf-modifer-key-down-crash.html:
```diff
@@ -7,7 +7,12 @@
+    testRunner.queueLoadingScript(`
+        (async function() {
+            setTimeout("history.back()", 0);
+            await testRunner.runUIScript("uiController.keyDown('', ['leftCommand'])", () => {});
```
Huh, didn't realize this function started returning a promise.

smfr replied:
Oh, it didn't (can we return Promises from native code yet?). This patch is bad.

rniwa replied:
We can but I don't think we have hooks wired up in WKTR.

rniwa replied:
@Constellation

pxlcoder replied:
I think we can use `UIHelper.keyDown` here?

smfr committed:
fast/events/ios/pdf-modifer-key-down-crash.html leaves an unfinished UIScript running
https://bugs.webkit.org/show_bug.cgi?id=243332
<rdar://98164628>

Reviewed by Ryosuke Niwa.

Use async/await in the queued script so that it waits for the runUIScript() to complete.

Also unskip the test.

* LayoutTests/fast/events/ios/pdf-modifer-key-down-crash.html:

Canonical link: https://commits.webkit.org/253253@main
